adala/runtimes/_litellm.py

```diff
@@ -0,0 +1,537 @@
+import asyncio
+import logging
+from typing import Any, Dict, List, Optional
+
+import httpx
+import litellm
+from adala.utils.internal_data import InternalDataFrame
+from adala.utils.logs import print_error
+from adala.utils.matching import match_options
+from adala.utils.parse import parse_template, partial_str_format
+from openai import NotFoundError
+from pydantic import ConfigDict, Field
+from rich import print
+from tenacity import retry, stop_after_attempt, wait_random_exponential
+
+from .base import AsyncRuntime, Runtime
+
+logger = logging.getLogger(__name__)
+
+
+@retry(wait=wait_random_exponential(min=1, max=60), stop=stop_after_attempt(6))
+async def async_create_completion(
+    model: str,
+    user_prompt: str,
+    client: httpx.Client,
+    system_prompt: Optional[str] = None,
+    api_key: Optional[str] = None,
```

Comment:
not actually used here, still have some cleaning to do...

Reply:
How does litellm handle retries? does it have a better way to respect rate limits built in?

Reply:
If this works, it could resolve our ticket about rate limiting... https://docs.litellm.ai/docs/routing#advanced---routing-strategies-%EF%B8%8F

Reply:
saw that, was hoping the same but have not had the chance to test. Probably going to need to test in a separate ticket at this point, but seems promising